plugin/handler_socket/handlersocket/database.cpp

```diff
@@ -304,6 +304,7 @@ dbcontext::init_thread(const void *stack_bottom, volatile int& shutdown_flag)
       thd->db = 0;
       thd->db = my_strdup("handlersocket", MYF(0));
     }
+    thd->variables.option_bits |= OPTION_TABLE_LOCK;
```

Comment:
mysql_unlock_tables() does not free sql_lock if this flag is not set. handlersocket expects it to be freed by mysql_unlock_tables().